src/main/modules/state-redirect/index.ts

```diff
@@ -46,8 +46,8 @@ export class StateRedirectMiddleware {
         }
 
         if (
-          req.session.userCase?.state !== State.AwaitingPayment ||
-          [PAY_YOUR_FEE, PAY_AND_SUBMIT, PAYMENT_CALLBACK_URL].includes(req.path as PageLink)
+          req.session.userCase?.state !== State.AwaitingPayment // ||
+          //[PAY_YOUR_FEE, PAY_AND_SUBMIT, PAYMENT_CALLBACK_URL].includes(req.path as PageLink)
```

Comment:
I think PAYMENT_CALLBACK_URL check should be there
you can remove the other 2 PAY_YOUR_FEE and PAY_AND_SUBMIT